resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -975,4 +984,45 @@ public class ApiClient implements AutoCloseable {
         @Override
         public void close() throws Exception { }
     }
+
+    private ApiClientConnectorResponse retryInvoke(ApiClientConnectorRequest connectorRequest, ApiClientConnectorResponse connectorResponse) {
+        int retries = 0;
+        long waitTimeInSec = 1L;
+        boolean retry;
+        int statusCode;
+        int MAX_RETRIES = 0;
+        long MAX_WAIT_INTERVAL_IN_SEC = 32L;
+
+        try {
+            final int retry_count = properties.getProperty(ApiClientConnectorProperty.MAX_RETRIES, Integer.class, 0);
+            if (retry_count > 0) {
+                MAX_RETRIES = retry_count;
+            } else {
+                return connectorResponse;
+            }
+
+            do {
+                connectorResponse = connector.invoke(connectorRequest);
+                statusCode = connectorResponse.getStatusCode();
+                retry = statusCode >= 300;
+                if (retry) {
+                    if (statusCode == 429) {
+                        if (connectorResponse.getHeaders().containsKey("Retry-After")) {
+                            waitTimeInSec = Integer.parseInt(connectorResponse.getHeaders().get("Retry-After"));
+                        }
+                    } else {
+                        waitTimeInSec = Math.min(getWaitTimeExp(retries), MAX_WAIT_INTERVAL_IN_SEC);
```

Comment:
The first few retries should honor the retry-after header if it's populated. Then fallback to a back-off if the failures continue.